### PR TITLE
Add function to tsdb.point to get line-protocol string in the correct…

### PR DESCRIPTION
**Pull request by sparrc**
… units

@jwilder @otoolep @corylanou

### CODE REVIEW

File: tsdb/points.go [hunk 1]
```diff
@@ -1159,6 +1160,9 @@ func (p *point) GetPrecisionMultiplier(precision string) int64 {
 	return int64(d)
 }
 
+// String returns a string representation of the point object, if there is a
```
Comment (otoolep) [inline tsdb/points.go]: Minor: would these comments be better on the interface spec?

File: tsdb/points.go [hunk 2]
```diff
@@ -1159,6 +1160,9 @@ func (p *point) GetPrecisionMultiplier(precision string) int64 {
...
+// timestamp associated with the point then it will be returned at the default
```
Comment (otoolep) [inline tsdb/points.go]: Minor: might be better phrased as "with the default precision of nanoseconds".

File: tsdb/points.go [hunk 3]
```diff
@@ -1166,6 +1170,17 @@ func (p *point) String() string {
 	return fmt.Sprintf("%s %s %d", p.Key(), string(p.fields), p.UnixNano())
 }
 
+// PrecisionString returns a string representation of the point object, if there
+// is a timestamp associated with the point then it will be returned at the given
```
Comment (otoolep) [inline tsdb/points.go]: "specified in the given unit"?

**Comment by pauldix** [comment]
I think it would be much better to have `SetPrecision` convert the given string into a `time.Duration`. Then when you're working with the timestamp to output it in the string, instead of doing division, use `.Round(p.Precision)` where precision is a duration.

It's cleaner and I think more readable.

**Comment by otoolep** [comment]
Can we see some unit tests? Would that make sense?

**Comment by sparrc** [comment]
@otoolep updated PR with unit tests and made some of a commenting changes you suggested

**Comment by sparrc** [comment]
@pauldix I'm not sure this API is working exactly as intended, so I'm going to have to leave in the division as it is right now, I'll send around an email for a separate discussion about what we should do about the write API

**Comment by otoolep** [comment]
+1, though I think @pauldix may still have some feedback pre-merge.

**Pull request merged**